### PR TITLE
update readme with components and workflow

### PR DESCRIPTION
**Pull request by hairongchen**
update readme with diagram and description of components and workflow to make clear what does the enclave-cc project look like for now and in near future. This part should be updated continually according to the project progress to make sure everyone involved in this project are on the same page.

Signed-off-by: hairongchen <hairong.chen@intel.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,9 +34,44 @@ from creating user container image with encryption/signature to uploading it to
...
+- LibOS: Occlum is supported and Gramine in future.
+- FUSE encrypted file system: enclave-agent puts decrypted app image into an encrypted fs and later referenced by LibOS as rootfs to start app enclave container.
+
+Deployment of enclave-cc runtime:
+- deploy of the operator according to [CoCo Operator installation guide](https://github.com/confidential-containers/operator/blob/main/docs/INSTALL.md)
+- install enclave-cc runtime according to [Getting started of CoCo](https://github.com/confidential-containers/documentation/blob/main/quickstart.md). This installation process will execute enclave-cc runtime [deployment script](https://github.com/confidential-containers/enclave-cc/blob/main/tools/packaging/deploy/enclave-cc-deploy.sh) to: 1) copy shim-rune and containerd binaries into host directories 2) copy image bundles of agent enclave container and boot instance to host directories 3) copy config of shim-rune to host directory 4) reconfigure containerd to enable enclave-cc runtime
+
+After the enclave-cc runtime is successfully installed by the operator, the application workload can be deployed. Bellow is the workflow of running an application workload:
```
Comment (dcmiddle) [inline README.md:58]:
```suggestion
After the enclave-cc runtime is successfully installed by the operator, the application workload can be deployed. Below is the workflow of running an application workload:
```

File: README.md [hunk 2]
```diff
@@ -34,9 +34,44 @@ from creating user container image with encryption/signature to uploading it to
...
+- end user defines a POD config yaml to describe the workload and runtime requirements according to  [the example](https://github.com/confidential-containers/documentation/blob/main/quickstart.md#creating-a-sample-coco-workload-using-enclave-cc)
+- end user deploys the POD into Kubernetes and request gets to containerd
```
Comment (dcmiddle) [inline README.md:60]:
```suggestion
- end user deploys the POD into Kubernetes and the request is propagated to containerd
```

File: README.md [hunk 3]
```diff
@@ -34,9 +34,44 @@ from creating user container image with encryption/signature to uploading it to
...
+- shim-rune gets sandbox and container creation request from containerd and creates pause container
+- shim-rune creates agent enclave container with runc; runc references OCI spec and roots in host preinstalled by operator to start the container with LibOS
```
Comment (dcmiddle) [inline README.md:62]:
```suggestion
- shim-rune creates agent enclave container with runc; runc references the OCI spec (config.json) and rootfs which were preinstalled on the host by the operator. This starts the container with a LibOS
```

File: README.md [hunk 4]
```diff
@@ -34,9 +34,44 @@ from creating user container image with encryption/signature to uploading it to
...
+- shim-rune asks enclave-agent to pull encrypted workload image via TTRPC request
+- enclave-agent receives image pull request, verifies the image signature by getting verification required resource from KBS after remote attestation
+- enclave-agent downloads the image and decrypts the workload image by getting decryption key from KBS
+- enclave-agent generates ephemeral key to encrypt the workload image with FUSE encrypted file system
+- shim-rune sends collected workload's config to enclave-agent
```
Comment (dcmiddle) [inline README.md:67]: I'm not sure what this means

File: README.md [hunk 5]
```diff
@@ -34,9 +34,44 @@ from creating user container image with encryption/signature to uploading it to
...
+- shim-rune creates app enclave container and runs LibOS boot init procedure
+- app enclave does local attestation with agent enclave and negotiates information encryption method(design/implementation ongoing)
+- enclave-agent sends workload's config in encryption to app enclave(design ongoing)
+- LibOS boot init procedure receives and decrypts the config; abstracts the fuse key to decrypt the FUSE encrypted file system(design ongoing)
+- LibOS boot init procedure mounts decrypted workload roots, builds workload running config and starts app enclave container
```
Comment (dcmiddle) [inline README.md:72]:
```suggestion
- LibOS boot init procedure mounts decrypted workload rootfs, builds workload running config and starts app enclave container
```

**Review by dcmiddle — CHANGES_REQUESTED** [review]
This is great content. 
It may belong in the design document under the architecture section. Consider placing it there.

Please also consider committing the source file for the image (e.g. powerpoint file). This will make it easier to maintain the diagram.

For markdown files please set line breaks at 100 characters or after each sentence. This makes future diffs easier.

**Comment by hairongchen** [comment]
> This is great content. It may belong in the design document under the architecture section. Consider placing it there.
> 
> Please also consider committing the source file for the image (e.g. powerpoint file). This will make it easier to maintain the diagram.
> 
> For markdown files please set line breaks at 100 characters or after each sentence. This makes future diffs easier.

Thanks Dan for all the feedbacks, I've updated the readme according to your suggestions and added the original ppt file. But for the location of the doc, I'd suggest we put it in readme directly which can catch everyone's attention easily and quick understand how enclave-cc actually works.

File: README.md [hunk 6]
```diff
@@ -34,9 +34,44 @@ from creating user container image with encryption/signature to uploading it to
...
+[Design document](docs/design.md)
+
+
+## Components and workflow
+Accoding to the design and implementation of this project, enclave-cc uses components from cloud native technology stack, Intel SGX technology stack and existing key mamagement technology stack together with its own developed components to deliver a process based confidential container solution in the Kubernetes platform with standard cloud native user experience.
+
+![ecc_components_workflow.png](docs/images/ecc_components_workflow.png)
+
+Where:
+- enclave-agent: enclave-agent is a process inside LibOS inside the enclave. It accepts request from shim-rune and handles image management, attestation, config management, and fuse encrypted file system management.
```
[the remark this answers is not in the document]
Comment (hairongchen) [reply README.md]: Hi Dan, I think the agent-enclave stands for an enclave  and enclave-agent stands for a process running in the enclave, and in this context, I think enclave-agent is more suitable.

File: README.md (hunk 4 above)
Comment (hairongchen) [reply README.md]: this refers to https://github.com/confidential-containers/enclave-cc/issues/20 and https://github.com/confidential-containers/enclave-cc/issues/46, which is not finalized yet but looking for a short term solution(after LA is ready), I put current proposed solution in to the diagram, but should be updated according to future actual implementation

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (dcmiddle) [reply README.md]: Hi Hairong, That's interesting. I hadn't interpreted it like that. That might be too nuanced for most people to understand... e-a != a-e  :-D 
At any rate I don't see that as blocking feedback for this PR, but maybe something we can sort out for consistency later.

**Review by dcmiddle — APPROVED** [review]

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (hairongchen) [reply README.md]: Agreed, we can later unify the terms or event make a term list when necessary.

**Comment by mkbhanda** [comment]
A bunch of minor suggestions! Good work!

**Comment by hairongchen** [comment]
> A bunch of minor suggestions! Good work!

Malini, thank you very much for the comments and diagram update suggestions! I've updated accordingly.

**Pull request merged**